GroupECommerce/MainForm.cs

```diff
@@ -24,11 +24,30 @@ private void addProductBtn_Click(object sender, EventArgs e)
 		}
 
         private void MainForm_Load(object sender, EventArgs e)
+        {
+            populateProductComboBox();
+        }
+
+        private void populateProductComboBox()
         {
             List<Product> allProducts = ProductDb.GetAllProducts();
 
             allProductsCbo.DataSource = allProducts;
             allProductsCbo.DisplayMember = nameof(Product.Title);
         }
+
+        private void deleteProductBtn_Click(object sender, EventArgs e)
+        {
+            Product currProduct = allProductsCbo.SelectedItem as Product;
+            DialogResult result = MessageBox.Show($"Are you sure you want to delete {currProduct.Title} for ${currProduct.Price} from the database?",
+                            "Warning",
+                            MessageBoxButtons.YesNo,
+                            MessageBoxIcon.Warning);
+            if(result == DialogResult.Yes)
+            {
+                ProductDb.Delete(currProduct);
+                populateProductComboBox();
```

Comment:
Do you possibly just want to remove the product from the combo box without completely reloading all elements. From what I know entity framework isn't the fastest.

Reply:
That is a way better idea I didn't think of that, I will edit the needed code.